### ISSUE TITLE
Edit megamenu content

### ISSUE THREAD

**Issue by VictoriaSunNIH** [NCIOCPL/cgov-digital-platform#3322]
## As a site admin, I want to be able to edit the megamenu contents so that users can find the content they need via our megamenu options.
> **ESTIMATE** TBD

As part of changing out the Mega Menu for NCIDS, we no longer want users to edit the raw HTML. There have been issues in the past where extra spaces were left in, or the HTML was malformed, and this leads to a broken site. We would like to use structured data to define the items in the megamenu, this data will be consumed by the front-end code as JSON. We will treat the Mega Menu Content as a block_content entity so that we can leverage all the management interfaces, translation and workflow without having to roll our own as we would in an Entity. Editors will select the mega menu content like they do today as a field on the `cgov_site_section`.

In a perfect word we would have client-side tool that allows users to layout the mega menu and select the links for it -- however that is out of scope for now, and might never be worth the LOE. So we can iteratively build up this functionality though. In discussions with IAs, manually editing JSON will be problematic as there is a high potential for errors. YAML would provide a more human-friendly version of JSON. Also as YAML can be converted to JSON and vice-versa it will work in our backend. Finally we can define a JSONSchema that would be used to validate the YAML. In a more perfect world, we could use a JavaScript code editor (e.g., Ace, Monoco, etc) to give a nicer user interface and error highlighting. However, this friendlier UI could be an add on.

### Tasks
1. Create the "NCIDS Mega Menu Content" block_content as [described in the wiki](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Block-Type%3A-NCIDS-Mega-Menu-Content)
2. Modify the cgov_site_section term type to add the NCIDS Mega Menu Contents field as [described in the wiki](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Vocabularies-Site-Sections#ncids-megamenu-contents)
    * This field should be added to the Form Display configuration (i.e., the editing page) beneath the current mega menu contents field.

---
### Acceptance Criteria

```gherkin
Scenario: User saves invalid YAML and form gives error
  Given [context]
    And [some more context]
  When  [event]
  Then  [outcome]
    And [another outcome]

Scenario: User saves valid YAML and is able to re-edit it.
  Given [context]
    And [some more context]
  When  [event]
  Then  [outcome]
    And [another outcome]
```

### Prerequisite Tickets
- #3337

**Comment by bryanpizzillo** [comment]
# Content Model Updates

## 1. Create a new block content type
---
### Properties
* **Display Name:** NCIDS Mega Menu Content
* **Machine-readable Name:** ncids_mega_menu_content

### Fields

#### Yah-mel Content
- **Field Name:**  field_yaml_content
- **Label:** Yaml Content
- **Type:** ❓ ❓ 
- **Validation Schema**: ❓:question: 
- **Required:** Yes
- **Tooltip Text:** N/A
- **Cardinality:** 1
- **Translatable:** No

## Form Display Order
- Yaml Content

## 2. Update the `cgov_site_section` to add a new field
---
#### NCIDS Mega Menu Contents
* **Field Name:** field_ncids_mega_menu_content
* **Label:** NCIDS Megamenu Contents
* **Type:** Entity Reference
  * Allows:  ncids_mega_menu_content
* **Required:** No
* **Validation Rules:** None
* **Cardinality:** 1
* **Translatable:** 
NOTE: Form Display Order between megamenu contents and display options

**Comment by bryanpizzillo** [comment]
Closed by #3397
